### PR TITLE
[reconfig] Finish end-to-end reconfig test

### PR DESCRIPTION
**Pull request by lxfind**
This PR fixes a few issues in the reconfiguration end-to-end test and add the flow of adding a new validator:
1. Increases the validator name limit. We were hitting that limit
2. Uses shared object transaction instead of owned object transaction

### CODE REVIEW

File: crates/sui-core/src/epoch/reconfiguration.rs [hunk 1]
```diff
@@ -168,6 +170,8 @@ where
                 "Error when processing advance epoch transaction: {:?}", err
             );
             tokio::time::sleep(WAIT_BETWEEN_EPOCH_TX_QUERY_RETRY).await;
+            cnt += 1;
+            assert!(cnt <= 50);
```
Comment (asonnino) [inline crates/sui-core/src/epoch/reconfiguration.rs:174]: Why do we need this?

File: crates/test-utils/src/authority.rs [hunk 2]
```diff
@@ -119,60 +114,3 @@ pub fn test_authority_aggregator(
...
-            .unwrap();
-        responses.push(reply);
-    }
-    responses
-}
-
-/// Keep submitting the certificates of a shared-object transaction until it is sequenced by
-/// at least one consensus node. We use the loop since some consensus protocols (like Tusk)
-/// may drop transactions. The certificate is submitted to every Sui authority.
-pub async fn submit_shared_object_transaction(
```
Comment (asonnino) [inline crates/test-utils/src/authority.rs:145]: Are we sure we don't need this anymore?

**Review by asonnino — APPROVED** [review]

File: crates/sui-core/src/epoch/reconfiguration.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-core/src/epoch/reconfiguration.rs:174]: This is temporary for debugging a test failure. Will remove once tests pass

File: crates/test-utils/src/authority.rs (hunk 2 above)
Comment (lxfind) [reply crates/test-utils/src/authority.rs:145]: Yes there is a duplicate in transaction.rs

**Pull request merged**